.github/workflows/build.yml

```diff
@@ -83,29 +83,34 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
+        pkg:
+          - ./client
+          - ./cmd/buildctl
+          - ./worker/containerd
+          - ./solver
+          - ./frontend
+          - ./frontend/dockerfile
+        worker:
+          - containerd
+          - containerd-1.4
+          - containerd-snapshotter-stargz
+          - oci
+          - oci-rootless
+          - oci-snapshotter-stargz
         include:
-          -
-            pkg: ./client
+          - pkg: ./client
```

Comment:
I'm not sure I understand how this interacts with the `matrix.pkg` or how gateway test only runs once.

Reply:
You can add additional configuration options to a build matrix job that already exists with `include`. In the case of the gateway:

```yaml
- pkg: ''
  skip-integration-tests: 1
  typ: integration gateway
```

it will be a "standalone" job because `pkg: ''` is not available in the current matrix.